org-code-javabuilder/lib/src/test/java/org/code/neighborhood/GridFactoryTest.java

```diff
@@ -0,0 +1,98 @@
+package org.code.neighborhood;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+public class GridFactoryTest {
+  String sampleGrid =
+      "[[\n{\"tileType\": 1}, {\"tileType\": 1}], \n[{\"tileType\": 1}, {\"tileType\": 1, \"value\": 4}]]";
+
+  @Test
+  void createGridFromString() {
+    GridFactory gridFactory = new GridFactory();
+    Grid grid = gridFactory.createGridFromString(sampleGrid);
+    assertTrue(grid instanceof Grid);
+    assertTrue(grid.validLocation(1, 1));
+  }
+
+  @Test
+  void createGridFromStringWithInvalidJSONThrowsException() {
+    GridFactory gridFactory = new GridFactory();
+    Exception exception =
+        assertThrows(
+            UnsupportedOperationException.class,
+            () -> {
+              gridFactory.createGridFromString("not valid json here:");
+            });
+    String expectedMessage = "Please check the format of your grid.txt file";
```

Comment:
All of our error messages are going to be specified on the client side so they can be translated. Here and other places, its sufficient to check that the right exception type is thrown without checking the error string.

Reply:
Ok. Here it is. 

All of our error messages are going to be specified on the client side so they can be translated. Rather than adding exception messages here, go ahead an add an exception enum like this:
https://github.com/code-dot-org/javabuilder/pull/32/files#diff-588560c58930937a4accd4ce0b91566398194bff4efdff36953ed6acf997f14c
For now, use the enum.toString directly as the exception message. We'll interpret those keys on the client side as we're building out neighborhood there.